src/main/java/ch/njol/skript/sections/EffSecOpenInventory.java

```diff
@@ -0,0 +1,283 @@
+			return false;
+		}
+
+		if (open && hasSection())
+			trigger = loadCode(sectionNode, "open inventory", InventorySectionEvent.class);
+
+		return true;
+	}
+
+	@Override
+	protected TriggerItem walk(Event event) {
+		if (object != null) {
+			Inventory inventory = null;
+			Object o = object.getSingle(event);
+			if (o instanceof Inventory i) {
+				inventory = i;
+			} else if (o instanceof InventoryType inventoryType && inventoryType.isCreatable()) {
+				inventory = createInventory(inventoryType);
+			}
+			if (inventory == null)
+				return super.walk(event, false);
+
+			Player[] players = this.players.getArray(event);
+			if (players.length > 0 && trigger != null) {
+				InventorySectionEvent inventoryEvent = new InventorySectionEvent(inventory);
+				Object localVars = Variables.copyLocalVariables(event);
+				Variables.setLocalVariables(inventoryEvent, localVars);
+				TriggerItem.walk(trigger, inventoryEvent);
+				Variables.setLocalVariables(event, Variables.copyLocalVariables(inventoryEvent));
+				Variables.removeLocals(inventoryEvent);
+			}
+
+			for (Player player : players)
+				player.openInventory(inventory);
+
+		} else {
+			for (Player player : players.getArray(event)) {
+				if (!open) {
+					player.closeInventory();
+					continue;
+				}
+				switch (syntax) {
+					case ANVIL:
+						player.openAnvil(null, true);
+						break;
+					case CARTOGRAPHY:
+						player.openCartographyTable(null, true);
+						break;
+					case ENCHANTING:
+						player.openEnchanting(null, true);
+						break;
+					case GRINDSTONE:
+						player.openGrindstone(null, true);
+						break;
+					case LOOM:
+						player.openLoom(null, true);
+						break;
+					case SMITHING:
+						player.openSmithingTable(null, true);
+						break;
+					case STONECUTTER:
+						player.openStonecutter(null, true);
+						break;
+					case WORKBENCH:
+						player.openWorkbench(null, true);
+						break;
+				}
+			}
+		}
+		return super.walk(event, false);
+	}
+
+	@Override
+	public String toString(@Nullable Event event, boolean debug) {
+		if (object != null)
+			return "show " + object.toString(event, debug) + " to " + players.toString(event, debug);
+		if (open)
+			return "open " + syntax.name().toLowerCase(Locale.ENGLISH) + " to " + players.toString(event, debug);
+		return "close inventory of " + players.toString(event, debug);
+	}
+
+	public static @Nullable Inventory createInventory(InventoryType type) {
+		if (!type.isCreatable())
+			return null;
+		try {
+			return Bukkit.createInventory(null, type);
+		} catch (NullPointerException | IllegalArgumentException e) {
+			// Spigot forgot to label some InventoryType's as non creatable in some versions < 1.19.4
+			// So this throws NullPointerException aswell ontop of the IllegalArgumentException.
+			// See https://hub.spigotmc.org/jira/browse/SPIGOT-7301
+			Skript.error("You can't open a '" + Classes.toString(type) + "' inventory to players. It's not creatable.");
+		}
```

Comment:
I don't think we support less than 1.19.4 anymore so it might be okay to remove this.